packages/core/src/test/awsService/accessanalyzer/iamPolicyChecks.test.ts

```diff
@@ -0,0 +1,793 @@
+                IamPolicyChecksConstants.MissingReferenceDocError,
+                getResultCssColor('Error'),
+            ])
+        )
+        assert(executeCommandStub.notCalled)
+        assert(executeCustomPolicyChecksCommandStub.notCalled)
+    })
+
+    it('checkAccessNotGranted should handle Terraform Plan document type correctly', async function () {
+        const documentType = 'Terraform Plan'
+        const actions = 'action1 action2'
+        const resources = 'resource1 resource2'
+        const document = 'path/to/terraform.json'
+        const cfnParameterPath = 'path/to/cfn-params'
+
+        sandbox.stub(IamPolicyChecksWebview, 'editedDocumentFileName').value(document)
+        sandbox.stub(iamPolicyChecks, 'isTerraformPlan').returns(true)
+        sandbox.stub(iamPolicyChecks, 'isCloudFormationTemplate').returns(false)
+        executeCustomPolicyChecksCommandStub = sandbox.stub(fakePolicyChecksWebview, 'executeCustomPolicyChecksCommand')
+
+        await fakePolicyChecksWebview.checkAccessNotGranted(documentType, actions, resources, cfnParameterPath)
+
+        assert(
+            executeCustomPolicyChecksCommandStub.calledOnceWith({
+                command: 'tf-policy-validator',
+                args: [
+                    'check-access-not-granted',
+                    '--template-path',
+                    document,
+                    '--region',
+                    'us-east-1',
+                    '--config',
+                    `${globals.context.asAbsolutePath(defaultTerraformConfigPath)}`,
+                    '--actions',
+                    'action1action2',
+                    '--resources',
+                    'resource1resource2',
+                ],
+                cfnParameterPathExists: !!cfnParameterPath,
+                documentType,
+                checkType: 'CheckAccessNotGranted',
+            })
+        )
+        assert(onCustomPolicyCheckResponseFireSpy.notCalled)
+        assert(executeCommandStub.notCalled)
+    })
+
+    it('checkAccessNotGranted should handle CloudFormation document type correctly', async function () {
+        const documentType = 'CloudFormation'
+        const actions = 'action1 action2'
+        const resources = 'resource1 resource2'
+        const document = 'path/to/cloudformation.yaml'
+        const cfnParameterPath = 'path/to/cfn-params'
+
+        sandbox.stub(IamPolicyChecksWebview, 'editedDocumentFileName').value(document)
+        sandbox.stub(iamPolicyChecks, 'isTerraformPlan').returns(false)
+        sandbox.stub(iamPolicyChecks, 'isCloudFormationTemplate').returns(true)
+        executeCustomPolicyChecksCommandStub = sandbox.stub(fakePolicyChecksWebview, 'executeCustomPolicyChecksCommand')
+
+        await fakePolicyChecksWebview.checkAccessNotGranted(documentType, actions, resources, cfnParameterPath)
+
+        assert(
+            executeCustomPolicyChecksCommandStub.calledOnceWith({
+                command: 'cfn-policy-validator',
```

Comment:
can you download the actual tool in CI and actually invoke it, instead of stubbing this?

Reply:
I don't think we want to call the actual tool as it makes calls to an actual AWS service and requires credentials. Do we have a set of credentials we use for testing AWS services here? If we do, I think we can do it; the test will depend on the availability of the service in that case.

Reply:
currently our CI only has a Builder Id token, not IAM credentials